src/main/java/com/internship/socialnetwork/controller/UserController.java

```diff
@@ -1,51 +1,91 @@
 package com.internship.socialnetwork.controller;
 
+import com.internship.socialnetwork.dto.FriendshipDto;
 import com.internship.socialnetwork.dto.UserDto;
+import com.internship.socialnetwork.service.FriendshipService;
 import com.internship.socialnetwork.service.UserService;
 import lombok.RequiredArgsConstructor;
 import org.springframework.http.ResponseEntity;
 import org.springframework.web.bind.annotation.DeleteMapping;
 import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PatchMapping;
 import org.springframework.web.bind.annotation.PathVariable;
 import org.springframework.web.bind.annotation.PostMapping;
 import org.springframework.web.bind.annotation.PutMapping;
 import org.springframework.web.bind.annotation.RequestBody;
 import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestParam;
 import org.springframework.web.bind.annotation.RestController;
 
 import java.util.List;
 
+import static org.springframework.http.HttpStatus.CREATED;
+
 @RestController
 @RequiredArgsConstructor
 @RequestMapping("/api/v1/users")
 public class UserController {
 
     private final UserService userService;
 
+    private final FriendshipService friendshipService;
+
     @GetMapping
-    public ResponseEntity<List<UserDto>> getAll() {
+    public ResponseEntity<List<UserDto>> getAllUsers() {
         return ResponseEntity.ok(userService.getAllUsers());
     }
 
     @GetMapping("/{id}")
-    public ResponseEntity<UserDto> get(@PathVariable Long id) {
+    public ResponseEntity<UserDto> getUser(@PathVariable Long id) {
         return ResponseEntity.ok(userService.getUser(id));
     }
 
+    @GetMapping("/{id}/friends")
+    public ResponseEntity<List<FriendshipDto>> getFriends(@PathVariable Long id) {
+        return ResponseEntity.ok(friendshipService.getFriends(id));
+    }
+
+    @GetMapping("/{id}/friend-requests")
+    public ResponseEntity<List<FriendshipDto>> getFriendRequests(@PathVariable Long id, @RequestParam boolean sent) {
```

Comment:
Here you can have FriendshipStatus enum instead of boolean.So you can find them by status or all if status is null. And make the parameter optional so we are not forced to put the parameter with null value